### PR TITLE
docs: add concept document on transports

### PR DESCRIPTION
**Pull request by mrinalwadhwa**

### CODE REVIEW

File: documentation/concepts/transports.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+messages over a variety of transport protocols like TCP, UDP, Bluetooth,
+ZigBee, LoRaWAN etc.
+
+Ockam includes a set of secure communication protocols, like
+[secure channels](./secure_channels) that provide their functionality in a
+way that is agnostic to the underlying means of transporting messages.
+This is achieved using an abstract `Transport` interface.
+
+A concrete implementation of the `Transport` interface is called an Ockam
+Transport. Over time there will be many such implementations.
```
Comment (robinbudd) [inline documentation/concepts/transports.md]: Did we agree to call it an Ockam Transport? Or was the transport going to refer to the lower-level, encryption-agnostic layer with Ockam Channel be the secure encrypted channel?

Comment (mrinalwadhwa) [reply documentation/concepts/transports.md]: transport is the the low level simple abstraction

**Comment by mrinalwadhwa** [comment]
I was trying to imply the opposite, so this needs improvement. Please read the secure channels description as well and let's see how we can communicate the intent better.
> This concepts doc seems to imply that the securing of data happens at the transport layer, which it doesn't.

File: documentation/concepts/transports.md [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+messages over a variety of transport protocols like TCP, UDP, Bluetooth,
+ZigBee, LoRaWAN etc.
+
+Ockam includes a set of secure communication protocols, like
+[secure channels](./secure_channels) that provide their functionality in a
+way that is agnostic to the underlying means of transporting messages.
+Such higher level protocols need a simple consistent mechanism, to send
+and receive messages, that only loosely couples them with the underlying
+transport protocol. This loose coupling is achieved using an abstract
+`Transport` interface.
```
Comment (robinbudd) [inline documentation/concepts/transports.md]: Nailed it!

File: documentation/concepts/transports.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
 ```yaml
 title: Transports
 ```
+
+Machines, within IoT systems, communicate amongst each other by exchanging
```
Comment (mattgreg) [inline documentation/concepts/transports.md]: Strike ',within IoT systems, '

I think that 'Machines' or 'Connected machines and devices' is a cleaner open

File: documentation/concepts/transports.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+transport protocol. This loose coupling is achieved using an abstract
+`Transport` interface.
+
+A concrete implementation of the `Transport` interface is called an Ockam
+Transport. Over time there will be many such implementations.
+
+![Transports](./assets/transports/transports.svg)
+
+This loose coupling between a higher level protocol and how a specific
+transport sends or receives messages will allow us to design secure protocols
```
Comment (mattgreg) [inline documentation/concepts/transports.md]: strike 'will allow'

'allows'

File: documentation/concepts/transports.md [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+and receive messages, that only loosely couples them with the underlying
+transport protocol. This loose coupling is achieved using an abstract
+`Transport` interface.
+
+A concrete implementation of the `Transport` interface is called an Ockam
+Transport. Over time there will be many such implementations.
+
+![Transports](./assets/transports/transports.svg)
+
+This loose coupling between a higher level protocol and how a specific
```
Comment (mattgreg) [inline documentation/concepts/transports.md]: ...between Ockam's higher level protocols and...

**Pull request merged**